src/components/AssetsTable/AssetsTable.vue

```diff
@@ -0,0 +1,317 @@
+        v-show="totalPages !== 1"
+        class="pagination-pages__number"
+      >
+        {{ totalPages }}
+      </button>
+      <button
+        @click="goToPage(currentPage + 1)"
+        :disabled="currentPage === totalPages"
+        class="pagination-pages__arrow"
+      >
+        {{ '>' }}
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import { fetchItemsByCategory } from '@/api/asset-management.api'
+import { getColumnNames } from '@/utils/getColumnNames'
+
+export default {
+  components: 'AssetsTable',
+  props: {
+    defaultCategory: {
+      type: String,
+      required: true,
+    },
+  },
+
+  created() {
+    this.getDataByCategory(this.defaultCategory)
+  },
+
+  watch: {
+    async defaultCategory(newCategory, oldCategory) {
+      if (newCategory !== oldCategory) {
+        this.currentPage = 1
+        this.getDataByCategory(this.defaultCategory)
+      }
+    },
+  },
+
+  methods: {
+    async getDataByCategory(defaultCategory) {
+      try {
+        const categoryData = await fetchItemsByCategory(defaultCategory)
+
+        if (categoryData[0] && typeof categoryData[0] === 'string') {
+          this.columns = categoryData.map(item => ({ name: item, label: item }))
+          this.rows = []
+          return
+        }
+
+        this.columns = getColumnNames(categoryData[0])
+        this.rows = categoryData
+        this.totalItems = categoryData.length
+        this.updatePagination()
+      } catch (err) {
+        throw new Error('Error during fetching the data', err)
+      }
+    },
+
+    sortTable(columnLabel) {
+      const { column, direction } = this.sortConfig
+
+      if (column === columnLabel) {
+        this.sortConfig.direction = direction === 'asc' ? 'desc' : 'asc'
+      } else {
+        this.sortConfig.column = columnLabel
+        this.sortConfig.direction = 'asc'
+      }
+      this.updatePagination()
+    },
+
+    getSortIcon(label) {
+      if (this.sortConfig.column === label) {
+        return this.sortConfig.direction === 'asc'
+          ? 'pi pi-sort-amount-up'
+          : 'pi pi-sort-amount-down'
+      }
+
+      return 'pi pi-sort-alt'
+    },
+
+    goToPage(pageNumber) {
+      if (pageNumber >= 1 && pageNumber <= this.totalPages) {
+        this.currentPage = pageNumber
+        this.updatePagination()
+      }
+    },
+
+    updatePagination() {
+      const sortedRows = this.sortRows(this.rows)
+      this.totalPages = (this.rows.length % this.rowsPerPage) + 1
+      this.paginationRows = this.getPaginatedRows(sortedRows)
+    },
+
+    getPaginatedRows(sortedRows) {
+      const startIndex = (this.currentPage - 1) * this.rowsPerPage
+      const endIndex = startIndex + this.rowsPerPage
+      return sortedRows.slice(startIndex, endIndex)
+    },
+
+    sortRows(rows) {
+      if (!this.sortConfig.column) return rows
+
+      const { column, direction } = this.sortConfig
+      const sortedList = [...rows].sort((a, b) => {
```

Comment:
you can use toSorted: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted

Reply:
looks like you have sortRows method and `sortedRows` computed property that are seemingly doing the same thing. Could you please explain this a bit?

Reply:
> looks like you have sortRows method and `sortedRows` computed property that are seemingly doing the same thing. Could you please explain this a bit?

I modified it and removed the second one. Previously first one was used to initialization mostly and second one was used for updating the vue. Originally I added sortedRows cuz some bug appeared after adding pagination. Changing pages caused table to lose sort ability after click.